presto-main/src/main/java/com/facebook/presto/operator/project/PageProcessor.java

```diff
@@ -103,6 +105,17 @@ public PageProcessorOutput process(ConnectorSession session, Page page)
         return projections;
     }
 
+    private static long calculatePageRetainedSizeWithoutLoading(Page page)
+    {
+        long retainedSizeInBytes = 0;
+        for (Block block : page.getBlocks()) {
+            if (!(block instanceof LazyBlock) || ((LazyBlock) block).isLoaded()) {
```

Comment:
What if we changed `LazyBlock.getRetainedSizeInBytes()` to return zero if the block is not loaded?